src/metrics/remote_peer_info.go

```diff
@@ -0,0 +1,84 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/protolambda/zrnt/eth2/beacon"
+)
+
+// BASIC HOST INFO
+
+// BasicHostInfo contains the basic Host info that will be requested from the identification of a libp2p peer
+type BasicHostInfo struct {
+	TimeStamp time.Time
+	// Peer Host/Node Info
+	PeerID          string
+	NodeID          string
+	UserAgent       string
+	ProtocolVersion string
+	Addrs           string
+	PubKey          string
+	RTT             time.Duration
+	Protocols       []string
+	// Information regarding the metadata exchange
+	Direction string
+	// Metadata requested
+	MetadataRequest bool
+	MetadataSucceed bool
+}
+
+// BEACON METADATA
+
+// Basic BeaconMetadata struct that includes the timestamp of the received beacon metadata
+type BeaconMetadataStamped struct {
+	Timestamp time.Time
+	Metadata  beacon.MetaData
+}
+
+// Generate a New BeaconMetadata struct and timestamp it
+// NOTE: It will append empty metadatas timestamped to track when did we request them
+func NewBMetadataStamped(meta beacon.MetaData) BeaconMetadataStamped {
```

Comment:
shouldnt this be part of the peer.go?
perhaps we dont need a constructor for this, as its easier to just `BeaconMetadataStamped{t.Now(), meta)`

Reply:
I will remove the constructor yep, looks cleaner.